conn.go

```diff
@@ -59,23 +69,109 @@ func (cc *Conn) Flush() error {
 
 	defer conn.Close()
 
-	if _, err := conn.SendMessages(batch(cc.messages)); err != nil {
+	cc.endBatch(cc.messages)
+
+	_, err = conn.SendMessages(cc.messages[:cc.it+1])
+
+	if err != nil {
 		return fmt.Errorf("SendMessages: %w", err)
 	}
 
-	if _, err := conn.Receive(); err != nil {
-		return fmt.Errorf("Receive: %w", err)
+	// Retrieving of seq number associated to entities
+	entitiesBySeq := make(map[uint32]Entity)
+	for i, e := range cc.entities {
```

Comment:
Instead of the separate `entites` book-keeping, why not just iterate over cc.messages and check if the type implements HandleResponse?

```
for _, msg := range cc.messages {
  m, ok := msg.(interface { HandleResponse(netlink.Message) })
  if !ok {
    continue
  }
  m.HandleResponse(rmsg)
}
```

Reply:
HandleResponse is not a method that belong to netlink.Message. It's belong to entity.
For a given entity, it's perform some operations on it by using the netlink response. We have to connect an entity (like a Rule) and a netlink message in case of the slice of responses is unordered. I connect them with the netlink sequence.

Reply:
Okay. I’m still not particularly happy about having to manage `messages` and `entities`, which are separate but connected.

Instead of the current approach, we could make `messages` take an interface type which has a `NetlinkMessage` method. putMessages would then wrap a `netlink.Message` in a type which implements `NetlinkMessage`, and can optionally implement `HandleResponse` as well.

Reply:
I'm not sure to understand what do you have in mind but the main limit I have is that the SendMessages method of netlink library take a slice of netlink.Message and not a slice of netlink.Message's pointer. So the only data struct that give me the netlink sequence number is the slice of netlink.Message returned by sendMessages. So the slice of netlink.Message have to be connected in one way or another to something else. In your solution I have to connected the type that wrap netlink.Message with the netlink.Message that contains the seq even if both are the same netlink message.

Reply:
I’m a little confused now: you’re saying the sequence numbers are only set in the []netlink.Message that’s returned by `(*netlink.Conn).SendMessages`, but in your PR, you’re discarding that result:
```go
if _, err = conn.SendMessages(cc.messages); err != nil {
```
Is that a bug in your PR, or am I misunderstanding?

(Independently, related to your question: the fact that the netlink package takes a []netlink.Message instead of []*netlink.Message doesn’t limit us in which data structures we want to use. If need be, we can do a copy from one to the other. A netlink.Message’s Header is just a few ints, and copying the Body byte slice won’t need to copy its contents.)